### PR TITLE
treewide: switch to texlive.withPackages

### PR DESCRIPTION
**Pull request by xworld21**
## Description of changes
Remove almost all calls to `texlive.combine` or `texlive.combined.scheme-*` with equivalent `texliveSmall.withPackages` calls – including new `texliveFull`, etc. at the top level. Rationale:
- a drop in evaluation time (I hope!)
- better ergonomics: `.withPackages` is composable, so dependencies can now live in the package definition instead of all-packages.nix, and users can easily override `texliveSmall` to pass a different TeX Live environment

There may bugs caused by TeX Live packages having the same name as a Nixpkgs package in scope, e.g. `with ps; [ texinfo ]` can accidentally refer to `pkgs.texinfo` instead of `ps.texinfo`. Hopefully I spotted all of them but please verify.

Not sure if we can fit this in time with 23.11 – it may cause a large rebuild. Let's see what ofborg says.

## Things done
- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by veprbl — APPROVED** [review]
Diff LGTM

**Comment by xworld21** [comment]
@GrahamcOfBorg build tests.texlive

**Comment by xworld21** [comment]
@GrahamcOfBorg build tests.texlive

**Comment by xworld21** [comment]
@GrahamcOfBorg build tests.texlive

(I can't do a proper review on my machine because `ghostscriptX` is still broken on darwin - it requires #263833.)

File: pkgs/tools/graphics/asymptote/default.nix [hunk 1]
```diff
@@ -33,7 +33,7 @@ stdenv.mkDerivation rec {
     ghostscriptX imagemagick fftw
     boehmgc ncurses readline gsl libsigsegv
     zlib perl curl qtbase qtsvg boost
-    texLive
+    (texliveSmall.withPackages (ps: with ps; [ epsf cm-super texinfo media9 ocgx2 collection-latexextra ]))
```
Comment (xworld21) [inline pkgs/tools/graphics/asymptote/default.nix:36]:
```suggestion
    (texliveSmall.withPackages (ps: with ps; [ epsf cm-super ps.texinfo media9 ocgx2 collection-latexextra ]))
```
Force push in a moment.

**Comment by xworld21** [comment]
@GrahamcOfBorg build tests.texlive

**Comment by xworld21** [comment]
The tests were passing on the last run, so they should still pass now:
@GrahamcOfBorg build tests.texlive

**Comment by xworld21** [comment]
As far as I can tell, the Darwin tests fail only because ghostscriptX is broken, so this is good to go.

**Comment by veprbl** [comment]
@GrahamcOfBorg build tests.texlive

**Review by veprbl — APPROVED** [review]

**Pull request merged**